src/crawlers.ts

```diff
@@ -0,0 +1,138 @@
+    startCrawler: boolean = true,
+) {
+    const proxyConfig = await Actor.createProxyConfiguration(crawlerOptions.proxyConfigurationOptions);
+
+    const crawler = new CheerioCrawler({
+        keepAlive: true,
+        proxyConfiguration: proxyConfig,
+        maxRequestRetries: 4,
+        requestQueue: queueSearchCrawler,
+        requestHandler: async ({ request, $: _$ }: CheerioCrawlingContext<UserData>) => {
+            // NOTE: we need to cast this to fix `cheerio` type errors
+            const $ = _$ as CheerioAPI;
+
+            log.info(`${CrawlerType.CHEERIO_GOOGLE_SEARCH_CRAWLER} requestHandler: Processing URL: ${request.url}`);
+            const organicResults = scrapeOrganicResults($);
+
+            let searchUrls = organicResults.map((result) => result.url).filter((url): url is string => url !== undefined);
+            // limit the number of search results to the maxResults
+            searchUrls = searchUrls.slice(0, request.userData?.maxResults ?? searchUrls.length);
+            log.info(`Extracted ${searchUrls.length} URLs: \n${searchUrls.join('\n')}`);
+
+            const responseId = request.uniqueKey;
+            for (const url of searchUrls) {
+                const r = createRequest(url, responseId);
+                await addContentCrawlRequest(r, crawlerOptions, scraperSettings);
+            }
+        },
+    });
+    if (startCrawler) {
+        crawler.run().then(() => log.warning(`Google-search-crawler has finished`), () => {});
+        log.info('Google-search-crawler has started 🫡');
+    }
+    const key = CrawlerType.CHEERIO_GOOGLE_SEARCH_CRAWLER;
+    crawlers.set(key, crawler);
+    return crawler;
+}
+
+export async function createAndStartCrawlerPlaywright(
+    crawlerOptions: CrawlerOptions = {} as CrawlerOptions,
+    settings: ScraperSettings = {} as ScraperSettings,
+    startCrawler: boolean = true,
+) {
+    const options: PlaywrightCrawlerOptions = {
+        ...(crawlerOptions as PlaywrightCrawlerOptions),
+        keepAlive: true,
+        requestQueue: await RequestQueue.open(),
+    };
+
+    const crawler = new PlaywrightCrawler({
+        requestHandler: (context: PlaywrightCrawlingContext) => genericHandler(context, settings),
+        ...options,
+    });
+
+    if (startCrawler) {
+        crawler.run().then(() => log.warning(`Crawler playwright has finished`), () => {});
+        log.info('Crawler playwright has started 💪🏼');
+    }
+    const key = CrawlerType.PLAYWRIGHT_CONTENT_CRAWLER;
+    crawlers.set(key, crawler);
+    return crawler;
+}
+
+/**
+ * Adds a search request to the Google search crawler.
+ * Create a response for the request and set the desired number of results (maxResults).
+ */
+export const addSearchRequest = async (
+    request: RequestOptions<UserData>, response: ServerResponse | null,
+    crawlerOptions: CrawlerOptions, maxResults: number,
+) => {
+    const key = CrawlerType.CHEERIO_GOOGLE_SEARCH_CRAWLER;
```

Comment:
I would add a comment that currently this getting of crawler from map by the key is not doing anything since we pre-create the two crawlers anyway but later it can be used e.g. if user wants to use different proxy configs